lib/connection.js

```diff
@@ -118,7 +118,7 @@ Connection.prototype.processPayload = function processPayload (payload) {
         if (id !== undefined && id !== null && typeof id !== 'string' && typeof id !== 'number') {
             return this.sendError('invalidRequest', id, { info: 'id, if provided, must be one of: null, string, number' });
         }
-        if (params !== null && typeof params !== 'object') {
+        if (params !== undefined && params !== null && typeof params !== 'object') {
             return this.sendError('invalidRequest', id, { info: 'params must be one of: null, object, array' });
```

Comment:
I didn't change it because the `id` error above doesn't specify `undefined`.  Do you want both changed?